.github/workflows/documentation.yml

```diff
@@ -8,8 +8,77 @@ on:
   workflow_dispatch:
 
 jobs:
+  generate-files:
```

Comment:
Have you tested the action?

if so and it's working, I'm happy to approve this.

Reply:
I have tested it [here](https://github.com/moj-analytical-services/splink/actions/runs/4241498663/jobs/7371797035) - the only differences are:
* using `mkdocs build` rather than `mkdocs gh-deploy`, as didn't want to update the live site, and seems there isn't away to avoid the 'push' step of the latter without fiddling with `mkdocs'` internals
* to get the `build` step to run I had to skip cloning of `splink_demos`
However these steps shouldn't affect the changed part of the workflow, and inspecting the output of `tree site/` in the 'deploy' step of that run shows that the new html files were generated, so believe the real workflow should work.